centreon/www/widgets/src/centreon-widget-resourcestable/src/Listing/Actions/Check/index.tsx

```diff
@@ -0,0 +1,168 @@
+import { PrimitiveAtom } from 'jotai';
+import { useTranslation } from 'react-i18next';
+
+import IconForcedCheck from '@mui/icons-material/FlipCameraAndroidOutlined';
+import IconCheck from '@mui/icons-material/Sync';
+
+import { Method, useMutationQuery } from '@centreon/ui';
+
+import { Resource } from '../../models';
+import { labelCheck, labelForcedCheck } from '../../translatedLabels';
+import { checkEndpoint } from '../api/endpoint';
+import { Data } from '../model';
+import ResourceActionButton from '../ResourceActionButton';
+import useAclQuery from '../aclQuery';
+
+import Check from './Check';
+import CheckOptionsList from './CheckOptionsList';
+import { CheckActionAtom, checkActionAtom } from './checkAtoms';
+import { adjustCheckedResources } from './helpers';
+import useStateCheckAction from './useStateCheckAction';
+
+interface Props {
+  resources: Array<Resource>;
+  stateCheckActionAtom?: PrimitiveAtom<CheckActionAtom | null>;
+  testId: string;
+}
+
+const CheckActionButton = ({
+  resources,
+  testId,
+  stateCheckActionAtom = checkActionAtom,
+  ...rest
+}: Props & Data): JSX.Element => {
+  const { t } = useTranslation();
+
+  const { checkAction, setCheckAction } = useStateCheckAction({
+    resources,
+    stateCheckActionAtom
+  });
+
+  const { onSuccessCheckAction, onSuccessForcedCheckAction } =
+    rest.successCallback || {};
+
+  const { mutateAsync: checkResource } = useMutationQuery({
+    getEndpoint: () => checkEndpoint,
+    method: Method.POST
+  });
+
+  const { canForcedCheck, canCheck } = useAclQuery();
+
+  const disableCheck = !canCheck(resources);
+  const disableForcedCheck = !canForcedCheck(resources);
+  const hasSelectedResources = resources.length > 0;
+
+  const isCheckPermitted = canCheck(resources) || !hasSelectedResources;
+
+  const isForcedCheckPermitted =
+    canForcedCheck(resources) || !hasSelectedResources;
+
+  const handleCheckResource = (): void => {
+    checkResource({
+      payload: {
+        check: { is_forced: false },
+        resources: adjustCheckedResources({ resources })
+      }
+    }).then(() => {
+      onSuccessCheckAction?.();
+    });
+  };
+
+  const handleForcedCheckResource = (): void => {
+    checkResource({
+      payload: {
+        check: { is_forced: true },
+        resources: adjustCheckedResources({ resources })
+      }
+    }).then(() => {
+      onSuccessForcedCheckAction?.();
+    });
+  };
+
+  const onClickCheck = (): void => {
+    setCheckAction({ checked: true, forcedChecked: false });
```

Comment:
a little proposition
wouldn't it be better and simpler to make checkAction just a variable that takes just 3 variables 'checked' | 'forceChecked' | 'undefined' in place of an object (because in any situation we will have just one of these, non ??)

But you know better, is it just a proposition ??

Reply:
I agree, but I used the same code as on the resource status page and did not create specific tests for this functionality. I relied on the tests for the resource status and want to avoid making significant changes to prevent altering the functionality